### DISCUSSION TITLE
Event on page closed for stop a watchdog

### DISCUSSION THREAD

**Discussion by fjosegalan** [reflex-dev/reflex#3378] [Q&A]
Hello

I am playing around with reflex and the test I am doing is as follow:

Before start the reflex APP I call a socket from back end to provide some data to states. 
Then in the main page load event "on_load" I call an app state method to enable a watchdog with the socket, to conditionally render some message in the page about socket not connected.

App start:
```python
from rxconfig import config
from .main import main
from .state import AppState
import reflex as rx

filename = f"{config.app_name}/{config.app_name}.py"

@rx.page(on_load=AppState.enable_watchdog)
def index() -> rx.Component:
    return  rx.hstack(
                main()
            )

app = rx.App()
app.add_page(index)
```

The App state looks like this:
```python
import asyncio
import reflex as rx
from .config import SIOMANAGER

class AppState(rx.State):
    _watchdog: bool = False
    sup_sioConnected : bool = False
    _n_tasks: int = 0`

    @rx.background
    async def background_task(self):
        async with self:
            if self._n_tasks > 0:
                return
            self._n_tasks +=1
        while True:
            print("bucle")
            async with self:
                if SIOMANAGER.IsConnected:
                    self.sup_sioConnected = True
                    self._n_tasks -= 1
                    return
                else:
                    SIOMANAGER.open_connection()
                    print("Connect")
            await asyncio.sleep(1.0)
         return

    def enable_watchdog(self):
        self._watchdog = True
        return AppState.background_task
```
As you can see there is a background task checking for the connection, and if not connected tries to reconnect it every 1 second.

What is the problem?
I set some logs (basic print functions) and the background task starts when the page is loaded, it works as expected. But when the page is closed on browser I have noted that the background task still running forever, as stated even with the web browser closed.

First question:
Is my approach to watchdog the socket connection ok?
Then second question:
Is there an event about "closed page"? or "on closing" page? to stop the background event for the watchdodg.

Regards

Fran Galan

**Comment by fjosegalan** [comment]
Sorry for my code quoting.... I did not managed it properly

**Reply by Lendemor** [reply]
(I edited your original message to format the code properly 👍 )

**Comment by Lendemor** [comment]
We don't have a specific event that is triggered on page close.

However, a recent PR (https://github.com/reflex-dev/reflex/pull/3388) added a way to keep track of whether the client websocket is still connected, so you can use that for now to detect if your background task should stop or keep running.

**Comment by benedikt-bartscher** [comment]
related: https://github.com/reflex-dev/reflex/pull/2671
